qa/tasks/cephfs/test_volumes.py

```diff
@@ -1871,6 +1871,32 @@ def test_subvolume_group_exists_without_subvolumegroup_and_with_subvolume(self):
         ret = self._fs_cmd("subvolumegroup", "exist", self.volname)
         self.assertEqual(ret.strip('\n'), "no subvolumegroup exists")
 
+    def test_subvolume_group_rm_when_its_not_empty(self):
+        group = self._generate_random_group_name()
+        subvolume = self._generate_random_subvolume_name()
+
+        # create subvolumegroup
+        self._fs_cmd("subvolumegroup", "create", self.volname, group)
+        # create subvolume in group
+        self._fs_cmd("subvolume", "create", self.volname, subvolume, "--group_name", group)
+        # try, remove subvolume group
+        try:
+            self._fs_cmd("subvolumegroup", "rm", self.volname, group)
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.ENOTEMPTY:
+                raise
```

Comment:
can we also test the exception being printed? i.e. `cannot delete subvolume group {group} since it is not empty`? im not sure if CommandFailedError sends stderr

Reply:
As far as I know, `CommandFailedError` sends only `exitstatus` and the `command`.

Reply:
hmm `_fs_cmd()` uses `get_ceph_cmd_stdout()` which as the name suggests returns only `stdout`(and obviously the errno but omitting `stderr`). This might be fine if we say we only want to look at the errno and decide test success/failure but my inclination always has been to also test `stderr` if an exception is expected i.e. in this case `cannot delete subvolume group {group} since it is not empty`. For this i'd say create a new helper named `_fs_complete_cmd()` and make it return all - errno, stderr and stdout. This would not only enable you test the exception string here but is also a very versatile solution for writing test cases in future as well. 

```
    def _fs_complete_cmd(self, cmd_args: list, raise_on_error=False):
        return self.run_ceph_cmd(args=["fs"] + cmd_args, stdout=StringIO(),
                                 stderr=StringIO(), check_status=raise_on_error)
```
NOTE: you can keep using `*args` but i like this style more.

using above you can then do:

```
ret = self._fs_complete_cmd(["subvolumegroup", "rm", self.volname, group])
```
and then fetch `stderr` using `ret[0].stderr.getvalue()`

This would not only help you get the stderr but also help you suppress raising exceptions if it is needed. You can directly then compare `ret[0].returncode` without having the try-exception block.

Reply:
I think it's sufficient to check the errno here.